src/features/registry/registry.ts

```diff
@@ -207,13 +216,124 @@ const resolveImageId = multiCacheMemoizee(
 	},
 );
 
+const resolveImageLocation = multiCacheMemoizee(
+	async (
+		application: string,
+		release: string | undefined,
+		service: string | undefined,
```

Comment:
How about v so that's clear what each param holds?
```suggestion
		applicationSlug: string,
		releaseHashOrVersion: string | undefined,
		serviceName: string | undefined,
```